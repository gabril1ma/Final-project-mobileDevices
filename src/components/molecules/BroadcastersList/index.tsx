import React from "react";
import { StyleSheet, View } from "react-native";
import useAppContext from "../../../hooks/useAppContext";
import CheckBox from "../../atoms/CheckBox/CheckBox";

const BroadcastersList = () => {
  const {
    appState: { broadcasters },
    setAppState,
  } = useAppContext();

  const handleCheckBoadcaster = (index: number) =>
    setAppState((prev) => ({
      ...prev,
      broadcasters: broadcasters?.map((broadcaster) => {
        if (broadcaster?.name === broadcasters[index]?.name)
          return { ...broadcaster, selected: !broadcaster?.selected };
        return broadcaster;
      }),
    }));

  return (
    <View style={styles.container}>
      {broadcasters?.map((b, index) => (
        <CheckBox
          key={b?.name}
          label={b?.name}
          isChecked={b?.selected}
          onValueChange={() => handleCheckBoadcaster(index)}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 16,
  },
});

export default BroadcastersList;
