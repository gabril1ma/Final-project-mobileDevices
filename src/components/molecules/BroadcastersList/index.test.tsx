import { create, ReactTestRendererJSON } from "react-test-renderer";
import BroadcastersList from ".";

export type TTreeRenderer = ReactTestRendererJSON | ReactTestRendererJSON[];

const broadcasterMock = [
  {
    name: "CNN Brasil",
    selected: false,
    urlAPI: "",
  },
  {
    name: "Folha de SÃ£o Paulo",
    selected: false,
    urlAPI: "",
  },
  {
    name: "Estado de Minas",
    selected: false,
    urlAPI: "",
  },
];

jest.mock("../../../hooks/useAppContext", () => ({
  __esModule: true,
  default: () => ({
    appState: {
      broadcasters: broadcasterMock,
    },
    setAppState: () => jest.fn(),
  }),
}));

describe("<BroadcastersList />", () => {
  let tree: TTreeRenderer;
  let testTree: ReactTestRendererJSON;

  beforeEach(() => {
    tree = create(<BroadcastersList />).toJSON();
    testTree = tree[0] ?? tree;
  });

  it("should render component", () => {
    expect(testTree).toMatchSnapshot();
  });

  it("should have one children", () => {
    expect(testTree?.children?.length).toBe(3);
  });
});
