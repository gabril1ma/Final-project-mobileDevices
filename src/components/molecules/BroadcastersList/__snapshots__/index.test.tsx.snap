// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BroadcastersList /> should render component 1`] = `
<View
  style={
    {
      "marginTop": 16,
    }
  }
>
  <Text
    onPress={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderBottomColor": "#DFE4EA",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="checkbox"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "borderColor": "#657786",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 20,
              "width": 20,
            },
            {
              "height": 20,
              "margin": 12,
              "width": 20,
            },
            false,
            {
              "backgroundColor": undefined,
              "borderColor": "#DFE4EA",
            },
            undefined,
            false,
          ]
        }
      />
      <Text
        style={
          {
            "fontSize": 15,
          }
        }
      >
        CNN Brasil
      </Text>
    </View>
  </Text>
  <Text
    onPress={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderBottomColor": "#DFE4EA",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="checkbox"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "borderColor": "#657786",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 20,
              "width": 20,
            },
            {
              "height": 20,
              "margin": 12,
              "width": 20,
            },
            false,
            {
              "backgroundColor": undefined,
              "borderColor": "#DFE4EA",
            },
            undefined,
            false,
          ]
        }
      />
      <Text
        style={
          {
            "fontSize": 15,
          }
        }
      >
        Folha de SÃ£o Paulo
      </Text>
    </View>
  </Text>
  <Text
    onPress={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderBottomColor": "#DFE4EA",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="checkbox"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "borderColor": "#657786",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 20,
              "width": 20,
            },
            {
              "height": 20,
              "margin": 12,
              "width": 20,
            },
            false,
            {
              "backgroundColor": undefined,
              "borderColor": "#DFE4EA",
            },
            undefined,
            false,
          ]
        }
      />
      <Text
        style={
          {
            "fontSize": 15,
          }
        }
      >
        Estado de Minas
      </Text>
    </View>
  </Text>
</View>
`;
